### PR TITLE
Add preferred_username support (OIDC provider)

### PR DESCRIPTION
**Pull request by ffdybuster**
## Description

This PR adds a new claim, `preferred_username`. I added code for the OIDC provider to support it and tested it with dex (https://github.com/dexidp/dex/pull/1566).

## Motivation and Context

Since #282 doesn't look like it will get done anytime soon, and since I really need `preferred_username`, I created this PR to add this specific claim. I don't really know the other connectors next to OIDC (and can't test them), so I only added support in there. I don't think the PR will break any of the other providers (the claim is returned empty for them and thus the headers will not be added).

## How Has This Been Tested?

I've build the docker container and ran it in our system for almost a day without problems. Our system uses dex via OIDC provider, and is used from Nginx via `auth_request`.

## Checklist:

- [X] My change requires a change to the documentation or CHANGELOG.
- [X] I have updated the documentation/CHANGELOG accordingly.
- [X] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

File: oauthproxy.go [hunk 1]
```diff
@@ -350,6 +351,10 @@ func (p *OAuthProxy) redeemCode(host, code string) (s *sessionsapi.SessionState,
 		s.Email, err = p.provider.GetEmailAddress(s)
 	}
 
+	if s.PreferredUsername == "" {
+		s.PreferredUsername, err = p.provider.GetPreferredUsername(s)
```
Comment (JoelSpeed) [inline oauthproxy.go]: This error is currently unchecked, and, for most providers, will always error, can we add a check on this error similar to the one below for `s.User`

Comment (felixfontein) [reply oauthproxy.go]: Sure! I've added it in ae34f54c816f1e44985f867c4b271704038ba388.

**Comment by felixfontein** [comment]
(Rebased to resolve conflicts.)

**Comment by ffdybuster** [comment]
@JoelSpeed thanks a lot for reviewing and merging! :)

**Pull request merged**